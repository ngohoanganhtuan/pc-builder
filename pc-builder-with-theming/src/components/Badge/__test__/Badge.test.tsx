import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';

// Components
import { Badge } from '@components';

describe('Test [Badge] component', () => {
  test('Component [Badge] matches DOM Snapshot', () => {
    const component = renderer.create(<Badge content={1000} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Component [Badge] should have span element tag', () => {
    render(<Badge content={1000} />);
    expect(screen.getByText('1000').tagName).toEqual('SPAN');
  });

  test('Component [Badge] can have custom class', () => {
    render(
      <Badge
        content={1000}
        className={'custom-class'}
      />
    );
    expect(screen.getByText('1000').classList).toContain('custom-class');
  });
});
