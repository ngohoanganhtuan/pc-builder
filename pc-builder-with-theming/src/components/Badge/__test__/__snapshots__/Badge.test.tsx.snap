// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test [Badge] component Component [Badge] matches DOM Snapshot 1`] = `
<span
  className="rounded-pill bg-danger text-white px-2 "
>
  1000
</span>
`;
